nasa/harmony
# HARMONY-1526: Change harmony in a box, localstack, and postgres to only accept connections from localhost.

## Jira Issue ID
HARMONY-1526

## Description
Sets up harmony in a box, localstack, and postgres kubernetes services to only accept connections from localhost.

## Local Test Steps
1. `npm run build` to create the new harmony image
2. Change your .env file to be for harmony in a box (or remove it and create from scratch)
3. Use the new `bin/stop-harmony-and-services` script to delete your harmony resources
4. Run `bin/bootstrap-harmony`.
5. Submit a request and verify it succeeds (note unrelated to this change I'm finding I have to delete my query-cmr and service pods if they have been restarted at least once as reported by kubectl get pods -n harmony because they have received a termination message and so won't pick up any work)
6. Verify all ports are available on localhost (
```
nc -z localhost 3000
nc -z localhost 5432
nc -z localhost 4566
nc -z localhost 4572
nc -z localhost 4592
```
7. Get your IP address and verify the same set of ports are not available `nc -z <IP> 3000 ...`
8. Verify http://localhost:3000/cloud-access works.
9. `bin/stop-harmony-and-services` and start things up for development
10. Copy back in your main .env file
11. `bin/start-postgres-localstack`
12. `bin/restart-services`
13. Repeat tests to make sure your env still works.

## PR Acceptance Checklist
